readthedocs/api/v2/views/model_views.py

```diff
@@ -224,11 +225,25 @@ class VersionViewSet(DisableListEndpoint, UserSelectViewSet):
 class BuildViewSet(DisableListEndpoint, UserSelectViewSet):
     permission_classes = [APIRestrictedPermission]
     renderer_classes = (JSONRenderer, PlainTextBuildRenderer)
-    serializer_class = BuildSerializer
-    admin_serializer_class = BuildAdminSerializer
     model = Build
     filterset_fields = ('project__slug', 'commit')
 
+    def get_serializer_class(self):
+        """
+        Return the proper serializer for UI and Admin.
+
+        This ViewSet has a sligtly different pattern since we want to
+        pre-process the `command` field before returning it to the user, and we
+        also want to have a specific serializer for admins.
+        """
+        if self.request.user.is_staff:
+            # Logic copied from `UserSelectViewSet.get_serializer_class`
+            # and extended to check for GET method
+            if self.request.method == "GET":
+                return BuildAdminUISerializer
+            return BuildAdminSerializer
+        return BuildSerializer
```

Comment:
```suggestion
                return BuildAdminUISerializer #Staff read-only
            return BuildAdminSerializer # Staff write-only
        return BuildSerializer # Non-staff
```

There's probably a way to make this logic more clear, but the way it is now could definitely use some clarification.